QModManager/API/ModLoading/Internal/QMod.cs

```diff
@@ -0,0 +1,339 @@
+            get => _moddedGame;
+            private set
+            {
+                _moddedGame = value;
+                this.Game = $"{_moddedGame}";
+            }
+        }
+
+        public bool IsLoaded
+        {
+            get
+            {
+                if (this.PatchMethods.Count == 0)
+                    return false;
+
+                foreach (PatchMethod patchingMethod in this.PatchMethods.Values)
+                {
+                    if (!patchingMethod.IsPatched)
+                        return false;
+                }
+
+                return true;
+            }
+        }
+
+        public bool IsValid { get; private set; } = true;
+
+        public Assembly LoadedAssembly
+        {
+            get => _loadedAssembly;
+            set
+            {
+                _loadedAssembly = value;
+
+                AssemblyName assemblyName = _loadedAssembly.GetName();
+                this.AssemblyName = assemblyName.Name;
+                this.ParsedVersion = assemblyName.Version;
+            }
+        }
+
+        public Version ParsedVersion
+        {
+            get => _modVersion;
+            private set
+            {
+                _modVersion = value;
+                this.Version = _modVersion.ToStringParsed();
+            }
+        }
+
+        public Dictionary<PatchingOrder, PatchMethod> PatchMethods { get; } = new Dictionary<PatchingOrder, PatchMethod>();
+
+        public ICollection<string> DependencyCollection { get; private set; }
+        public ICollection<string> LoadBeforeCollection { get; private set; }
+        public ICollection<string> LoadAfterCollection { get; private set; }
+
+        public ModLoadingResults TryLoading(PatchingOrder order, Game currentGame)
+        {
+            if ((this.ParsedGame & currentGame) == API.Game.None)
+            {
+                this.PatchMethods.Clear(); // Do not attempt any other patch methods
+                return ModLoadingResults.CurrentGameNotSupported;
+            }
+
+            if (!this.PatchMethods.TryGetValue(order, out PatchMethod patchMethod))
+                return ModLoadingResults.NoMethodToExecute;
+
+            if (patchMethod.IsPatched)
+                return ModLoadingResults.AlreadyLoaded;
+
+            if (!patchMethod.TryInvoke())
+            {
+                this.PatchMethods.Clear(); // Do not attempt any other patch methods
+                return ModLoadingResults.Failure;
+            }
+
+            Logger.Info($"Loaded mod \"{this.Id}\" at {order}");
+            return ModLoadingResults.Success;
+        }
+
+        public bool TryCompletingJsonLoading(string subDirectory)
+        {
+            switch (this.Game)
+            {
+                case "BelowZero":
+                    _moddedGame = API.Game.BelowZero;
+                    break;
+                case "Both":
+                    _moddedGame = API.Game.Both;
+                    break;
+                case "Subnautica":
```

Comment:
The optimizer will remove it at compile time. This is here as documentation.

Alternatively, this should probably instead be doing more stricture checking to make sure that the provided text is correct instead of just defaulting to "Subnautica" when an invalid string is provided.